lib/reader.ml

```diff
@@ -66,8 +66,8 @@ let rec make =
       let result = { content  = String.make n '\000';
                      before   = "";
                      after    = String.make m '\000'; } in
-      Pervasives.really_input c result.content 0 n;
-      ignore @@ Pervasives.input c result.after 0 n;
+      Pervasives.really_input c Bytes.(of_string result.content) 0 n;
+      ignore @@ Pervasives.input c Bytes.(of_string result.after) 0 n;
```

Comment:
Same here: I don't think this works because it immediately throws away the `bytes` buffer created. You should allocate the buffers as bytes, do the input, then convert them to strings when creating the result. I think in this context we can use `Bytes.unsafe_to_string` to avoid copying, since we know this function is the unique owner of the buffers.